lia-ui/src/main/java/com/lithium/community/android/ui/components/adapters/LiConversationAdapter.java

```diff
@@ -169,136 +185,317 @@ private void showPopupMenu(final View view, final int position) {
+
+        MessageSubscriptionCallback(int position, LiMessage item) {
+            this.position = position;
+            this.item = item;
+        }
+
+        @Override
+        public void onSuccess(LiBaseRestRequest request, LiPostClientResponse response) throws LiRestResponseException {
+            actionInProgress(false);
+
+            if (response.getHttpCode() == HttpURLConnection.HTTP_OK || response.getHttpCode() == HttpURLConnection.HTTP_CREATED) {
+                LiUIUtils.showInAppNotification(activity, R.string.li_action_subscribe_successful);
+                activity.runOnUiThread(new Runnable() {
+                    @Override
+                    public void run() {
+                        item.getUserContext().setSubscribed(true);
+                        notifyItemChanged(position);
+                    }
+                });
+            } else {
+                LiUIUtils.showInAppNotification(activity, R.string.li_action_subscribe_unsuccessful);
+            }
+        }
+
+        @Override
+        public void onError(Exception exception) {
+            LiUIUtils.showInAppNotification(activity, R.string.li_action_subscribe_unsuccessful);
+            actionInProgress(false);
+            Log.d(LiSDKConstants.GENERIC_LOG_TAG, exception.getMessage());
+        }
+    }
+
+    private void unsubscribe(LiMessage item, int position) {
+        actionInProgress(true);
+
+        LiUser user = LiSDKManager.getInstance().getLoggedInUser();
+        LiClientRequestParams.LiUserMessageSusbscriptionRequestParans params
+                = new LiClientRequestParams.LiUserMessageSusbscriptionRequestParans(activity, "" + item.getId(), "" + user.getId());
+        try {
+            LiClient client = LiClientManager.getUserMessageSubscriptionsClient(params);
+            client.processAsync(new MessageUserSubscriptionsCallback());
+        } catch (LiRestResponseException lrre) {
+            actionInProgress(false);
+            LiUIUtils.showInAppNotification(activity, R.string.li_action_unsubscribe_unsuccessful);
+            Log.d(LiSDKConstants.GENERIC_LOG_TAG, lrre.getMessage());
+        }
+    }
+
+    public class MessageUserSubscriptionsCallback implements LiAsyncRequestCallback<LiGetClientResponse> {
+
+        @Override
+        public void onSuccess(LiBaseRestRequest request, LiGetClientResponse response) throws LiRestResponseException {
+            if (response.getHttpCode() == HttpURLConnection.HTTP_OK || response.getHttpCode() == HttpURLConnection.HTTP_CREATED) {
+                JsonObject responseObject = response.getJsonObject();
+                if (!responseObject.has("data")) {
+                    LiUIUtils.showInAppNotification(activity, R.string.li_action_unsubscribe_unsuccessful);
+                    actionInProgress(false);
+                    return;
+                }
+                JsonObject dataObject = responseObject.getAsJsonObject("data");
+                if (!dataObject.has("items")) {
```

Comment:
Can create constants for "data", "items", "id" ?

Reply:
sure